internal/dispatch/graph/reachableresources_test.go

```diff
@@ -603,3 +609,109 @@ func TestCaveatedReachableResources(t *testing.T) {
 		})
 	}
 }
+
+func TestReachableResourcesWithConsistencyLimitOf1(t *testing.T) {
+	defer goleak.VerifyNone(t, goleakIgnores...)
+
+	ctx, dispatcher, revision := newLocalDispatcherWithConsistencyLimit(t, 1)
+	defer dispatcher.Close()
+
+	target := ONR("user", "owner", "...")
+	stream := dispatch.NewCollectingDispatchStream[*v1.DispatchReachableResourcesResponse](ctx)
+	err := dispatcher.DispatchReachableResources(&v1.DispatchReachableResourcesRequest{
+		ResourceRelation: RR("folder", "view"),
+		SubjectRelation: &core.RelationReference{
+			Namespace: target.Namespace,
+			Relation:  target.Relation,
+		},
+		SubjectIds: []string{target.ObjectId},
+		Metadata: &v1.ResolverMeta{
+			AtRevision:     revision.String(),
+			DepthRemaining: 50,
+		},
+	}, stream)
+	require.NoError(t, err)
+
+	for range stream.Results() {
+		// Break early
+		break
+	}
+	dispatcher.Close()
+}
+
+func TestReachableResourcesMultipleEntrypointEarlyCancel(t *testing.T) {
```

Comment:
My assumption is that the leak originates from cancellation not being propagated.

Reply:
That and the possibility of deadlock causing things to just hang forever